src/osd/ReplicatedPG.cc

```diff
@@ -453,6 +453,17 @@ void ReplicatedPG::wait_for_degraded_object(const hobject_t& soid, OpRequestRef
   op->mark_delayed("waiting for degraded object");
 }
 
+void ReplicatedPG::block_write_on_full_cache(
+  const hobject_t& _oid, OpRequestRef op)
+{
+  const hobject_t oid = _oid.get_head();
+  dout(20) << __func__ << ": blocking object " << oid
+	   << " on full cache" << dendl;
+  assert(objects_blocked_on_cache_full.count(oid) == 0);
+  objects_blocked_on_cache_full.insert(oid);
```

Comment:
This assert will fail for the do_op caller when we are blocking additional ops on an already-blocked object...